django/contrib/admin/filters.py

```diff
@@ -416,3 +416,46 @@ class RelatedOnlyFieldListFilter(RelatedFieldListFilter):
     def field_choices(self, field, request, model_admin):
         pk_qs = model_admin.get_queryset(request).distinct().values_list('%s__pk' % self.field_path, flat=True)
         return field.get_choices(include_blank=False, limit_choices_to={'pk__in': pk_qs})
+
+
+class EmptyFieldListFilter(FieldListFilter):
+    def __init__(self, field, request, params, model, model_admin, field_path):
+        self.lookup_kwarg = '%s__empty' % field_path
+        self.lookup_val = params.get(self.lookup_kwarg)
+        super().__init__(field, request, params, model, model_admin, field_path)
+
+    def queryset(self, request, queryset):
+        if not self.used_parameters or self.lookup_kwarg not in self.used_parameters:
+            return queryset
+
+        del (self.used_parameters[self.lookup_kwarg])
+        empty = True if self.lookup_val == '1' else False
+
+        if not self.field.empty_strings_allowed:
+            self.used_parameters['%s__isnull' % self.field_path] = empty
+            return super().queryset(request, queryset)
+
+        # Special case for fields allowing empty strings
+        lookup_condition = models.Q(**{'%s__isnull' % self.field_path: True}) | models.Q(**{self.field_path: ''})
```

Comment:
The `isnull` only need to be added if `self.field.null` else it might result in unnecessarily inefficient queries.